src/server/server_impl.go

```diff
@@ -138,7 +140,20 @@ func (server *server) Start() {
 	if err != nil {
 		logger.Fatalf("Failed to open HTTP listener: '%+v'", err)
 	}
-	logger.Fatal(http.Serve(list, server.router))
+	server.listenerMu.Lock()
+	server.httpListener = list
+	server.listenerMu.Unlock()
+	err = http.Serve(list, server.router)
+
+	server.listenerMu.Lock()
+	explicitlyStopped := server.stopped
+	server.listenerMu.Unlock()
+
+	if !explicitlyStopped {
+		logger.Fatal(err)
```

Comment:
Just for my understanding, when does this happen? In the signal exit case, there is an exit(0) there, so that won't cause this, right? Is it basically just if somehow Serve() returns without us calling close()?

Reply:
I looked at http/server.go and I can make this cleaner by checking if the err is http.ErrServerClosed. I'll do that

The other reasons for an error would be some kind of permanent network error from the listener